experimental/zerotrie/src/builder/mod.rs

```diff
@@ -0,0 +1,140 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+mod branch_meta;
+pub(crate) mod bytestr;
+pub(crate) mod konst;
+#[cfg(feature = "litemap")]
+mod litemap;
+#[cfg(feature = "alloc")]
+pub(crate) mod nonconst;
+
+use bytestr::ByteStr;
+
+use super::ZeroTrieSimpleAscii;
+
+impl<const N: usize> ZeroTrieSimpleAscii<[u8; N]> {
+    /// **Const Constructor:** Creates an [`ZeroTrieSimpleAscii`] from a sorted slice of keys and values.
+    ///
+    /// This function needs to know the exact length of the resulting trie at compile time.
+    ///
+    /// Also see [`Self::from_sorted_str_tuples`].
+    ///
+    /// # Panics
+    ///
+    /// Panics if `items` is not sorted or if `N` is not correct.
+    ///
+    /// # Examples
+    ///
+    /// Create a `const` ZeroTrieSimpleAscii at compile time:
+    ///
+    /// ```
+    /// use zerotrie::ZeroTrieSimpleAscii;
+    ///
+    /// // The required capacity for this trie happens to be 17 bytes
+    /// const TRIE: ZeroTrieSimpleAscii<[u8; 17]> = ZeroTrieSimpleAscii::from_sorted_u8_tuples(&[
+    ///     (b"bar", 2),
+    ///     (b"bazzoo", 3),
+    ///     (b"foo", 1),
+    /// ]);
+    ///
+    /// assert_eq!(TRIE.get(b"foo"), Some(1));
+    /// assert_eq!(TRIE.get(b"bar"), Some(2));
+    /// assert_eq!(TRIE.get(b"bazzoo"), Some(3));
+    /// assert_eq!(TRIE.get(b"unknown"), None);
+    /// ```
+    ///
+    /// Panics if strings are not sorted:
+    ///
+    /// ```compile_fail
+    /// # use zerotrie::ZeroTrieSimpleAscii;
+    /// const TRIE: ZeroTrieSimpleAscii<[u8; 17]> = ZeroTrieSimpleAscii::from_sorted_u8_tuples(&[
+    ///     (b"foo", 1),
+    ///     (b"bar", 2),
+    ///     (b"bazzoo", 3),
+    /// ]);
+    /// ```
+    ///
+    /// Panics if capacity is too small:
+    ///
+    /// ```compile_fail
+    /// # use zerotrie::ZeroTrieSimpleAscii;
+    /// const TRIE: ZeroTrieSimpleAscii<[u8; 15]> = ZeroTrieSimpleAscii::from_sorted_u8_tuples(&[
+    ///     (b"bar", 2),
+    ///     (b"bazzoo", 3),
+    ///     (b"foo", 1),
+    /// ]);
+    /// ```
+    ///
+    /// Panics if capacity is too large:
+    ///
+    /// ```compile_fail
+    /// # use zerotrie::{ZeroTrieSimpleAscii, AsciiStr};
+    /// const TRIE: ZeroTrieSimpleAscii<[u8; 20]> = ZeroTrieSimpleAscii::from_sorted_u8_tuples(&[
+    ///     (b"bar", 2),
+    ///     (b"bazzoo", 3),
+    ///     (b"foo", 1),
+    /// ]);
+    /// ```
+    pub const fn from_sorted_u8_tuples(tuples: &[(&[u8], usize)]) -> Self {
+        use konst::*;
+        let byte_str_slice = ByteStr::from_byte_slice_with_value(tuples);
+        let result = ZeroTrieBuilderConst::<N>::from_tuple_slice::<100>(byte_str_slice);
```

Comment:
issue: please mention what the `100` does here and below

Reply:
Done